pingcap/tidb
# *: make load data atomic by default

Signed-off-by: Tina Fritz <tina77fritz@gmail.com>



### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:
`load data` uses several transactions by default which will break transaction atomic property.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:
If `tidb_dml_batch_size = 0`, `load data` will atomic and make it default.

### Related changes

### Check List 

Tests 

- Unit test
- Manual test
1. load a large file
1. see the log of the server
```
[2020/07/27 23:07:59.160 +08:00] [INFO] [server.go:388] ["new connection"] [conn=5] [remoteAddr=127.0.0.1:46754]
[2020/07/27 23:08:52.888 +08:00] [INFO] [2pc.go:285] ["[BIG_TXN]"] [con=5] ["table ID"=50] [size=3921322] [keys=135218] [puts=135218] [dels=0] [locks=0] [checks=0] [txnStartTS=418345787592015872]
[2020/07/27 23:08:52.892 +08:00] [INFO] [2pc.go:394] ["2PC detect large amount of mutations on a single region"] [region=44] ["mutations count"=135218]
[2020/07/27 23:08:53.022 +08:00] [INFO] [2pc.go:394] ["2PC detect large amount of mutations on a single region"] [region=49] ["mutations count"=135218]
[2020/07/27 23:08:53.029 +08:00] [INFO] [load_data.go:255] ["commit one task success"] [conn=5] ["commit time usage"=901.4976ms] ["keys processed"=135218] ["tasks processed"=1] ["tasks in queue"=0]
```

Side effects

- Performance regression
    - Consumes more MEM

### Release note 

- No release note
